invenio_deposit/receivers.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Deposit listeners."""
+
+from __future__ import absolute_import, print_function
+
+from invenio_indexer.tasks import index_record
+
+
+def index_deposit_after_publish(sender, action=None, pid=None, deposit=None):
+    """Index the record after publishing."""
+    if action == 'publish':
+        record_uuid = deposit.fetch_published()[1].id
```

Comment:
The resolver returns a pair ``(pid, record)``. Do you mean that the return value of fetch_published should be changed ?